src/main/java/org/opensearch/agent/tools/CreateAnomalyDetectorToolEnhanced.java

```diff
@@ -0,0 +1,1402 @@
+                        indexName,
+                        OUTPUT_KEY_CATEGORY_FIELD,
+                        categoryField,
+                        OUTPUT_KEY_AGGREGATION_FIELD,
+                        aggregationField,
+                        OUTPUT_KEY_AGGREGATION_METHOD,
+                        aggregationMethod,
+                        OUTPUT_KEY_DATE_FIELDS,
+                        fallbackDateField,
+                        "interval",
+                        interval
+                    );
+
+                nextPhaseCallback.accept(suggestions, listener);
+            }
+        });
+    }
+
+    // Checks if detector config is valid (fields exist, aggregations work, etc.)
+    private <T> void validateDetectorPhase(
+        Map<String, String> suggestions,
+        MappingContext mappingContext,
+        String tenantId,
+        int maxRetries,
+        int currentRetry,
+        ActionListener<T> listener
+    ) {
+        try {
+            log.info("Validating detector configuration");
+            AnomalyDetector detector = buildAnomalyDetectorFromSuggestions(suggestions);
+
+            callValidationAPI(detector, "detector", new ActionListener<ValidateConfigResponse>() {
+                @Override
+                public void onResponse(ValidateConfigResponse response) {
+                    if (response.getIssue() != null) {
+                        String errorMessage = response.getIssue().getMessage();
+                        String issueType = response.getIssue().getType().toString();
+
+                        // GENERAL_SETTINGS = max detectors reached for cluster
+                        if ("GENERAL_SETTINGS".equals(issueType)) {
+                            log.error("System limit error (non-retryable): {}", errorMessage);
+                            respondWithError(listener, suggestions.get(OUTPUT_KEY_INDEX), "validation", errorMessage);
+                            return;
+                        }
+                        if (currentRetry < MAX_DETECTOR_VALIDATION_RETRIES) {
+                            retryDetectorValidation(
+                                suggestions,
+                                mappingContext,
+                                errorMessage,
+                                tenantId,
+                                maxRetries,
+                                currentRetry + 1,
+                                listener
+                            );
+                        } else {
+                            log.error("Max detector validation retries reached: {}", errorMessage);
+                            listener
+                                .onFailure(
+                                    new RuntimeException(
+                                        "Detector validation failed after " + MAX_DETECTOR_VALIDATION_RETRIES + " retries: " + errorMessage
+                                    )
+                                );
+                        }
```

Comment:
This conditional (`currentRetry >= MAX_DETECTOR_VALIDATION_RETRIES`) should be a fast-fail at the top of the `validateDetectorPhase` method, otherwise you spend a lot of time validating the detector which, if validation succeeds, you suddenly can't use.